src/McBopomofo.cpp

```diff
@@ -853,6 +892,81 @@ bool McBopomofoEngine::handleCandidateKeyEvent(
     }
   }
 
+  if (keyHandler_->inputMode() == McBopomofo::InputMode::McBopomofo) {
+    auto* choosingCandidate =
+        dynamic_cast<InputStates::ChoosingCandidate*>(state_.get());
+    bool isPlusKey = key.check(FcitxKey_plus) || key.check(FcitxKey_equal);
+    bool isMinusKey =
+        key.check(FcitxKey_minus) || key.check(FcitxKey_underscore);
+    if (choosingCandidate != nullptr && (isPlusKey || isMinusKey)) {
+      int page = candidateList->currentPage();
+      int pageSize = candidateList->size();
+      int index = candidateList->cursorIndex();
+      int selectedCandidateIndex = page * pageSize + index;
+      auto candidate = choosingCandidate->candidates[selectedCandidateIndex];
+      std::string reading = candidate.reading;
+      // If the reading has an invalid prefix, skip dictionary lookup
+      if (std::any_of(invalidPrefixes.begin(), invalidPrefixes.end(),
+                      [&reading](const std::string& prefix) {
+                        return reading.compare(0, prefix.length(), prefix) == 0;
+                      })) {
+        return true;
+      }
+
+      // If the reading doesn't contain a hyphen, return true
+      if (reading.find('-') == std::string::npos) {
+        return true;
+      }
+      std::string phrase = candidate.value;
+      std::string originalValue = candidate.originalValue;
+      if (phrase != originalValue) {
```

Comment:
This will need a comment on why we bail on this condition. Also consider renaming all `originalValue` to `rawValue`.